### PR TITLE
[RD2] Fix duplicate Elevate payment on same day update

### PR DESCRIPTION
**Pull request by lparrott**
- Fix for bug that charges donor twice if Elevate Commitment is edited on day of payment.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/RD2_ScheduleService.cls [hunk 1]
```diff
@@ -531,7 +531,7 @@ public without sharing class RD2_ScheduleService {
     public Date getNextDonationDateGreaterThan(RecurringDonationSchedule__c schedule, Date referenceDate) {
         Date firstValid = getFirstPossibleDonationDateFromStart(schedule);
 
-        if (firstValid >= referenceDate) {
+        if (firstValid > referenceDate) {
```
Comment (npsp-reedestockton) [inline src/classes/RD2_ScheduleService.cls:534]: I have serious concerns about this.  At the very least it requires rigorous testing.

Comment (npsp-reedestockton) [reply src/classes/RD2_ScheduleService.cls:534]: I just tested this and it seems to work.  The only concern I had was around the first payment date, and that works as expected, so I think this is okay.

**Review by npsp-reedestockton — APPROVED** [review]

File: src/classes/RD2_ScheduleService.cls (hunk 1 above)
Comment (lparrott) [reply src/classes/RD2_ScheduleService.cls:534]: I had the same concern, it's honestly a little weird that it works, maybe we need a better understanding of `firstOccurrenceOnTimestamp` (that's where this Date ends up)

**Review by andrewyu-salesforce — APPROVED** [review]
Everything works with the new change on both same and future date update.

**Pull request merged**